ccloud-monitoring/docs/consumer-scenarios/consumer-lag-problem.rst

```diff
@@ -0,0 +1,164 @@
+.. _ccloud-monitoring-consumer-connectivity-problem:
+
+Consumer Lag Problem
+********************
+
+Consumer group lag is a tremendous performance indicator. It tells you the difference between the
+producer's last produced message and the consumer group's last commit. If you are unfamiliar with
+consumer groups or concepts like committing offsets, please refer to this
+`Kafka Consumer documentation <https://docs.confluent.io/platform/current/clients/consumer.html>`__.
+
+A large or quickly growing lag indicates that the consumer is not able to keep up with
+the volume of messages on a topic.
+
+This scenario will look at Confluent Cloud metrics from the Metrics API, `kafka-lag-exporter <https://github.com/lightbend/kafka-lag-exporter>`__ metrics, and
+client metrics from the client application’s MBean object ``kafka.consumer:type=consumer-fetch-manager-metrics,client-id=<client_id>``.
+
+Introduce failure scenario
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. Stop ``consumer-1`` container, thus removing a consumer from the consumer group and add 2 producers:
+
+   .. code-block:: bash
+
+      docker-compose up -d --scale consumer=1 --scale producer=5
+
+   Which will produce the following output:
+
+   .. code-block:: bash
+
+      ccloud-exporter is up-to-date
+      kafka-lag-exporter is up-to-date
+      node-exporter is up-to-date
+      grafana is up-to-date
+      prometheus is up-to-date
+      Stopping and removing ccloud-monitoring_consumer_2 ... done
+      Starting ccloud-monitoring_producer_1              ... done
+      Creating ccloud-monitoring_producer_2              ... done
+      Creating ccloud-monitoring_producer_3              ... done
+      Creating ccloud-monitoring_producer_4              ... done
+      Creating ccloud-monitoring_producer_5              ... done
+      Starting ccloud-monitoring_consumer_1              ... done
+
+Diagnose the problem
+^^^^^^^^^^^^^^^^^^^^
+
+#. Open `Grafana <localhost:3000>`__ and login with the username ``admin`` and password ``password``.
+
+#. Navigate to the ``Consumer Client Metrics`` dashboard. Wait 2 minutes and then observe:
+
+   - ``Rebalance rate`` (``rebalance-rate-per-hour``) has a bump, indicating that the consumer group ``demo-cloud-monitoring-1`` underwent a rebalance, which is to be expected when a consumer leaves the group.
+
+   |Consumer Rebalance Bump|
+
+   - An upward trend in ``Consumer group lag in records``.  ``Consumer group lag in seconds`` will have a less dramatic increase. Both indicating that the producer is creating more messages than the consumer can fetch in a timely manner. These metrics are derived from ``kafka-lag-exporter``.
```

Comment:
If ``kafka-lag-exporter`` is not an MBean, explain what is providing those stats (e.g. explain the docker container, what it does, how it works, etc -- not verbose, just a sentence or two)

Reply:
👍